File: 2792-neighboring-bitwise-xor/2792-neighboring-bitwise-xor.py
class Solution:
    def doesValidArrayExist(self, derived: List[int]) -> bool:
        xor_sum = 0
        for num in derived:
            xor_sum ^= num
        
        # if XOR sum is 0, a valid original array exists because of circular dependency
        return xor_sum == 0
        

File: 2792-neighboring-bitwise-xor/2792-neighboring-bitwise-xor.js
/**
 * @param {number[]} derived
 * @return {boolean}
 */
var doesValidArrayExist = function (derived) {

    let xorSum = 0;
    for(const num of derived){
        xorSum ^= num;
    }

     // If the XOR sum is 0, a valid original array exists
    return xorSum === 0;
};